sql/14.9/extra/postgis.sql

```diff
@@ -0,0 +1,604 @@
+SELECT gm, gm1 FROM "types_PostGIS" WHERE gm != decode('0101000020e6100000bf72ce99fe763e40ed4960730ed84d40', 'hex') IS NOT NULL;
+--Testcase 365:
+EXPLAIN (VERBOSE, COSTS OFF)
+SELECT gm, gm1 FROM "types_PostGIS" WHERE gm &&& decode('0101000020e6100000bf72ce99fe763e40ed4960730ed84d40', 'hex') IS NOT NULL;
+--Testcase 366:
+EXPLAIN (VERBOSE, COSTS OFF)
+SELECT gm, gm1 FROM "types_PostGIS" WHERE gm |=| decode('0101000020e6100000bf72ce99fe763e40ed4960730ed84d40', 'hex') IS NOT NULL;
+--Testcase 367:
+EXPLAIN (VERBOSE, COSTS OFF)
+SELECT gm, gm1 FROM "types_PostGIS" WHERE gm <#> decode('0101000020e6100000bf72ce99fe763e40ed4960730ed84d40', 'hex') IS NOT NULL;
+--Testcase 368:
+EXPLAIN (VERBOSE, COSTS OFF)
+SELECT gm, gm1 FROM "types_PostGIS" WHERE gm <<->> decode('0101000020e6100000bf72ce99fe763e40ed4960730ed84d40', 'hex') IS NOT NULL;
+--Testcase 369:
+EXPLAIN (VERBOSE, COSTS OFF)
+SELECT gm, gm1 FROM "types_PostGIS" WHERE gm @ decode('0101000020e6100000bf72ce99fe763e40ed4960730ed84d40', 'hex') IS NOT NULL;
+--Testcase 370:
+EXPLAIN (VERBOSE, COSTS OFF)
+SELECT gm, gm1 FROM "types_PostGIS" WHERE gm ~ decode('0101000020e6100000bf72ce99fe763e40ed4960730ed84d40', 'hex') IS NOT NULL;
+
+
+--Testcase 400:
+EXPLAIN (VERBOSE, COSTS OFF)
+SELECT gg, gg1 FROM "types_PostGIS" WHERE gg <-> gg1 IS NOT NULL;
+--Testcase 401:
+EXPLAIN (VERBOSE, COSTS OFF)
+SELECT gg, gg1 FROM "types_PostGIS" WHERE gg && gg1 IS NOT NULL;
+--Testcase 402:
+EXPLAIN (VERBOSE, COSTS OFF)
+SELECT gg, gg1 FROM "types_PostGIS" WHERE gg <> gg1 IS NOT NULL;
+--Testcase 403:
+EXPLAIN (VERBOSE, COSTS OFF)
+SELECT gg, gg1 FROM "types_PostGIS" WHERE gg <> gg1 IS NOT NULL;
+--Testcase 404:
+EXPLAIN (VERBOSE, COSTS OFF)
+SELECT gg, gg1 FROM "types_PostGIS" WHERE gg = gg1 IS NOT NULL;
+--Testcase 405:
+EXPLAIN (VERBOSE, COSTS OFF)
+SELECT gg, gg1 FROM "types_PostGIS" WHERE gg != gg1 IS NOT NULL;
+
+--Testcase 450:
+EXPLAIN (VERBOSE, COSTS OFF)
+SELECT gg, gg1 FROM "types_PostGIS" WHERE gg <-> decode('0101000020e6100000bf72ce99fe763e40ed4960730ed84d40', 'hex') IS NOT NULL;
+--Testcase 451:
+EXPLAIN (VERBOSE, COSTS OFF)
+SELECT gg, gg1 FROM "types_PostGIS" WHERE gg && decode('0101000020e6100000bf72ce99fe763e40ed4960730ed84d40', 'hex') IS NOT NULL;
+--Testcase 452:
+EXPLAIN (VERBOSE, COSTS OFF)
+SELECT gg, gg1 FROM "types_PostGIS" WHERE gg <> decode('0101000020e6100000bf72ce99fe763e40ed4960730ed84d40', 'hex') IS NOT NULL;
+--Testcase 453:
+EXPLAIN (VERBOSE, COSTS OFF)
+SELECT gg, gg1 FROM "types_PostGIS" WHERE gg <> decode('0101000020e6100000bf72ce99fe763e40ed4960730ed84d40', 'hex') IS NOT NULL;
+--Testcase 454:
+EXPLAIN (VERBOSE, COSTS OFF)
+SELECT gg, gg1 FROM "types_PostGIS" WHERE gg != decode('0101000020e6100000bf72ce99fe763e40ed4960730ed84d40', 'hex') IS NOT NULL;
+
+
+--Testcase 500:
+DROP FOREIGN TABLE "types_PostGIS";
+--Testcase 501:
+DROP EXTENSION sqlite_fdw CASCADE;
+--Testcase 502:
+DROP EXTENSION postgis CASCADE;
+--Testcase 503: -- types for later types test
+CREATE EXTENSION postgis;
```

Comment:
@mkgrgis 
Each test file should be independent, so that we can execute it alone without any problem.  
Therefore, please remove this test case, and create `postgis` extension in `types` test file if you want to use it there.

Reply:
@son-phamngoc , `types` test executes both with `nogis` and `postgis`. Anyway `types` test should give equal results. Does it mean I should move some TCs to `nogis` and `postgis` tests from `types`?

Reply:
`nogis` and `postgis` are different type of test, depending on the flag ENABLE_GIS. Therefore, please do not test them in the same test file.  
You can move some TCs to `nogis` and `postgis`, or create 2 files to `types`. Example: `types_nogis` and `type_postgis`.

Reply:
Ok, will be derived. Could you please also explain me why we have 2 testing sequences in `Makefile` and `test.sh` ?

Reply:
@son-phamngoc , fixed. Please verify.

Reply:
@mkgrgis I don't understand what 2 testing sequences that you mentioned.  
Makefile defines which test file will be tested using `REGRESS` variable.  
`test.sh` is a utility script to provide a convenient way to test all test files. It rewrites the `REGRESS` variable in Makefile, and then use Makefile to compile, install and test.  

About your fix, it is OK. Thank you. Please help me to resolve this comment.